src/components/item/ItemsContainer.js

```diff
@@ -8,133 +8,30 @@ import Item from 'components/item/Item';
 const ItemsContainer = () => {
     const { state, dispatch } = useStore();
     const {
-      size,
       cart,
       products,
       hasBeenFiltered,
       currentProducts
     } = state;
 
-    // Add selected item to customers cart
-    const addToCart = (item, numberOfItems) => {
```

Comment:
The #1 wrong thing about this implementation  was that `<Item />` was not responsible for adding itself to the cart. There's no need to props drill this function down, but you needed to because this function needed access to `size`. The solution is to make `<Item />` be a form that contains local state for size and quantity.